app/controllers/employees_controller.rb

```diff
@@ -1,34 +1,64 @@
 # frozen_string_literal: true
 
 class EmployeesController < ApplicationController
+  before_action :new_form, only: [:new, :edit]
+  before_action :list_employees, only: [:new, :index, :edit]
+  before_action :find_employee, only: [:edit, :update, :update_status]
+  after_action :close_form, only: [:create, :update]
   def index
-    @employees = Employee.all
   end
 
   def new
     @employee = Employee.new
+    render 'index'
   end
 
   def create
-    Employee.create name: params['name'], position: params['position'], email: ['email'],
-                    private_number: ['private_number']
+    name = params[:employee][:name].upcase
+    Employee.create name: name, role_id: params['role'], 
+                    email: params[:employee][:email], branch_id: params['branch'], 
+                    private_number: params[:employee][:private_number], status: true
```

Comment:
I think all the logic in this create and update methods is a little bit hard to understand, probably there are things that needs to live in the model instead of the controller.

And as well, I think it's more clear and easy to read the code the the functions actually return a value, instead of assigning @variables.

And last but not least important, it would be of a lot of help to understand in the future the output of this controller if you add a unitary test with rspec, and in that way if someone in the future modifies this controller or any of your models, the rspec will ensure that everything remains working as normal.

Reply:
Thanks, I'm going to make some changes

Reply:
I've added this changes to EmployeesController and i'm going to add some tests later. Thank you again.